app/ui/src/app/core/providers/api-http-provider.service.ts

```diff
@@ -45,59 +47,74 @@ export class ApiHttpProviderService extends ApiHttpService {
       post: <T>(body: any) => this.post<T>([endpointKey, ...endpointParams], body),
       put: <T>(body: any) => this.put<T>([endpointKey, ...endpointParams], body),
       delete: <T>() => this.delete<T>(url),
-      upload: <T>(fileList?: FileList, body?: StringMap<any>) => {
-        return this.upload<T>([endpointKey, ...endpointParams], fileList, body);
+      upload: <T>(fileMap?: FileMap, body?: StringMap<any>) => {
+        return this.upload<T>([endpointKey, ...endpointParams], fileMap, body);
       }
     };
   }
 
   get<T>(endpoint: string | any[]): Observable<T> {
     const { endpointKey, endpointParams } = this.deconstructEndpointParams(endpoint);
     const url = this.getEndpointUrl(endpointKey, ...endpointParams);
-    return this.httpClient.get<T>(url);
+    return this.httpClient
+      .get<T>(url)
+      .catch(error => Observable.throw(this.catchError(error)));
   }
 
   post<T>(endpoint: string | any[], body?: any): Observable<T> {
     const { endpointKey, endpointParams } = this.deconstructEndpointParams(endpoint);
     const url = this.getEndpointUrl(endpointKey, ...endpointParams);
-    return this.httpClient.post<T>(url, body);
+    return this.httpClient
+      .post<T>(url, body)
+      .catch(error => Observable.throw(this.catchError(error)));
   }
 
   put<T>(endpoint: string | any[], body: any): Observable<T> {
     const { endpointKey, endpointParams } = this.deconstructEndpointParams(endpoint);
     const url = this.getEndpointUrl(endpointKey, ...endpointParams);
-    return this.httpClient.put<T>(url, body);
+    return this.httpClient
+      .put<T>(url, body)
+      .catch(error => Observable.throw(this.catchError(error)));
   }
 
   delete<T>(endpoint: string | any[]): Observable<T> {
     const { endpointKey, endpointParams } = this.deconstructEndpointParams(endpoint);
     const url = this.getEndpointUrl(endpointKey, ...endpointParams);
-    return this.httpClient.delete<T>(url);
+    return this.httpClient
+      .delete<T>(url)
+      .catch(error => Observable.throw(this.catchError(error)));
```

Comment:
Interesting that we weren't catching those before..

Reply:
Actually the application is not very resilient to backend exceptions, so it breaks half of the times when something goes wrong upon requesting data.

By combining catch operators propagated from the ApiHttpService with catch operators in RxJS streams (usually located at the related Effects middleware class) which funnel the caught error to a a particular XXXX_FAIL action, we ensure the application is resilient and never gets in a bad state that prevents the user from keep browsing.

This also allows to leverage generic error components and display proper error messages in the components with minimum boilerplate, since all instances of state feature a common `hasErrors::Boolean` and `errors::Array<Error>` properties.